pages/getTitleAndContentByIndex/index.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { NextApiHandler } from 'next';
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { isValidNetwork } from '~src/api-utils';
+import { postsByTypeRef } from '~src/api-utils/firestore_refs';
+import messages from '~src/util/messages';
+import { MessageType } from '~src/auth/types';
+import { ProposalType, getProposalTypeTitle, getSubsquidProposalType } from '~src/global/proposalType';
+import { GET_PROPOSAL_BY_INDEX_AND_TYPE } from '~src/queries';
+import fetchSubsquid from '~src/util/fetchSubsquid';
+
+export interface ITitleAndContent {
+	title: string;
+	content: string;
+}
+
+export const getTitleAndContentByIndex = async ({ network, index }: { network: string; index: number }) => {
+	try {
+		const query = GET_PROPOSAL_BY_INDEX_AND_TYPE;
+		const postVariables = {
+			index_eq: Number(index),
+			type_eq: getSubsquidProposalType(ProposalType.REFERENDUM_V2)
+		};
+		const subsquidRes = await fetchSubsquid({
+			network,
+			query: query,
+			variables: postVariables
+		});
+		const proposalData = subsquidRes.data.proposals?.[0];
+		if (!proposalData) {
+			return {
+				data: { content: '', title: '' } as ITitleAndContent,
+				error: 'The Proposal number is not valid',
+				status: 400
+			};
+		}
+		const postDocRef = postsByTypeRef(network, ProposalType.REFERENDUM_V2).doc(String(index));
+		const data = await postDocRef.get();
+		if (data.exists) {
+			const post = data.data();
```

Comment:
you can also get title and description from subsqure if it is not exist in our firebase